Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -5698,14 +5683,21 @@ bool WebGLRenderingContextBase::validateTexFuncData(const char* functionName, Te
         synthesizeGLError(error, functionName, "invalid texture dimensions");
         return false;
     }
-    CheckedUint32 total = srcOffset;
-    total *= JSC::elementSize(pixels->getType());
-    total += totalBytesRequired;
-    total += skipBytes;
-    if (total.hasOverflowed() || pixels->byteLength() < total) {
+
+    CheckedSize totalImageBytes = totalBytesRequired;
```

Comment:
This is part of the rabbit hole mentioned elsewhere.
The validate function is used in cases where the size is `GLuint` and in cases size is `GLisize`. You now regress the part which uses `GLuint`

Reply:
I was trying to standardise on doing the arithmetic with `size_t` and then checking that the result fits into the range of `GLuint` or `GLsizei` at the point we go to one of those APIs. Would you prefer to just always use `GLsizei`?